chainer/training/extensions/snapshot_writers.py

```diff
@@ -57,6 +57,28 @@ def save(self, filename, outdir, target, savefun, **kwds):
             savefun(tmppath, target)
             shutil.move(tmppath, os.path.join(outdir, filename))
 
+        self._post_save()
+
+    def _add_cleanup_hook(self, hook_fun):
+        """Adds cleanup hook function.
+
+        Technically, arbitrary user-defined hook can be called, but
+        this is intended for cleaning up stale snapshots.
+
+        Args:
+            hook_fun (callable): callable funtion to be called
+                right after save is done. It takes no arguments.
+
+        """
+        if not hasattr(self, '_post_save_hooks'):
```

Comment:
How about initializing it with `_post_save_hooks = []` in `Writer.__init__`?

Reply:
I didn't do that probably because  Writer didn't have `__init__()` method, but will do that.